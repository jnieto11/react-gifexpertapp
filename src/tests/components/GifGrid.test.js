import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import '@testing-library/jest-dom';
import { GifGrid } from '../../components/GifGrid';
import { shallow, configure } from 'enzyme';

configure({ adapter: new Adapter() });
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid/>', () => {

    const category = 'One Punch';

    test('Debe mostrarse correctamente ', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    })



    test('Debe mostar items cuando se cargan imagenes con useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            title: 'https://localhost/cualquiercosa.jpg',
            url: 'Cualquier cosa'
        },
        {
            id: 'ABC1',
            title: 'https://localhost/cualquiercosa.jpg',
            url: 'Cualquier cosa'
        }];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    })

})

