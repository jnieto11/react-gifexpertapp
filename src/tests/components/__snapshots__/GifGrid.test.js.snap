// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid/> Debe mostar items cuando se cargan imagenes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="https://localhost/cualquiercosa.jpg"
      url="Cualquier cosa"
    />
    <GifGridItem
      id="ABC1"
      key="ABC1"
      title="https://localhost/cualquiercosa.jpg"
      url="Cualquier cosa"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid/> Debe mostrarse correctamente  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
